### DISCUSSION TITLE
java8 and sdk java.time package

### DISCUSSION THREAD

**Discussion by yubinkim444** [ical4j/ical4j#673] [Q&A]
The current issue is arising in an Android project when using the iCal4j library, due to a conflict between Java 8's java.time package and the java.time package in the Android SDK. 
The iCal4j library internally uses Java 8's java.time package, leading to a collision. This results in a java.lang.NoSuchMethodError during runtime.
How can I resolve this error?

This is my lib version
            library("calendar-ical4j", "org.mnode.ical4j:ical4j:4.0.0-rc1")
            library("calendar-ical4jExt", "org.mnode.ical4j:ical4j-extensions:2.0.0-rc1")
            library("calendar-threetenabp","com.jakewharton.threetenabp:threetenabp:1.3.1")

**Comment by benfortuna** [comment]
I am not sure if anyone has tried using ical4j 4.x on Android yet, but hopefully there is a solution. 

Note that we do make use of some classes in `java.time` in ical4j 3.x (i.e. [TemporalAmount](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/temporal/TemporalAmount.html?is-external=true)), and I believe this has been working on Android successfully for a while now.
